src/averageAges.js

```diff
@@ -15,11 +15,20 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person =>
+    century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm'
+  );
+
+  function functionAge() {
+    const ages = men.reduce((sum, age) => sum + (age.died - age.born), 0);
+    const agesMiddle = ages / men.length;
+
+    return agesMiddle;
+  }
```

Comment:
move it out of this function so other functions have accsess to it. And avoid duplication